packages/rum/src/domain/segmentCollection/segment.ts

```diff
@@ -71,7 +71,12 @@ export class Segment {
   }
 
   addRecord(record: Record): void {
-    this.metadata.end = record.timestamp
+    if (record.timestamp < this.metadata.start) {
+      this.metadata.start = record.timestamp
+    }
+    if (this.metadata.end < record.timestamp) {
+      this.metadata.end = record.timestamp
+    }
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏it could be easier to read:
```suggestion
    this.metadata.start = Math.min(this.metadata.start, record.timestamp)
    this.metadata.end = Math.max(this.metadata.end, record.timestamp)
```